### ISSUE TITLE
[Feature Request] New OnAddingToSelection Event

### ISSUE THREAD

**Issue by LordNullPointer** [JAM-Software/Virtual-TreeView#635]
VirtualTreeView should have a "OnAddingToSelection Event" to have the chance to interrupt the selection process. Many events do have these kinds of event e.g. OnFocusChanging, OnHeaderDragging, OnExpanding.

It would be great to get this event to disallow the selection of specific nodes. Useful for example if there're some nodes which should only act as a seperator for a category.

**Comment by joachimmarder** [comment]
> It would be great to get this event to disallow the selection of specific nodes.

Nodes that cannot be selected should be made disabled using the property `IsDisbaled[]`.

**Comment by LordNullPointer** [comment]
> Nodes that cannot be selected should be made disabled using the property IsDisbaled[].

Hmm that's an interesting point. But I think it's not a good way to disable a complete node only because I don't want to select it. Maybe in the next step I want to edit this node. First step would then be to enable this node again and then start editing. I think that's not the best way.

Is there a reason to not implement this event? I think it would be a really good helper event for a lot of cases.

**Comment by sanjayssk** [comment]
Once you disallow selection on a node, there is no user interface paradigm possible in the tree to specify that node as the target of any further action like editing. It has to be some other application interface and logic for the user to be able to specify that node as a target  for further actions. That same logic can reenable the node if necessary, for example, before calling EditNode on it.

On the other hand, if you don't want the user to add to an existing selection then perhaps, you can do that in onFocusChanging by disallowing those nodes to get focus once a selection is made.

**Comment by joachimmarder** [comment]
> Maybe in the next step I want to edit this node.

But this requires that the user first selects this specific node, and you won't know that the user want to edit the node when he tries to select it. of course this problem applies to both approaches but isn't easier to solve in one of them.

> Is there a reason to not implement this event?

Possible reasons:
- It does not really add a new feature, it should be able to achieve the same using `IsDiabled[]`and `OnAddToSelection`
- Maintainability. When I took over the development of Virtual TreeView I found that it is pretty difficult to maintain, it is very monolithic and contains a lot of copied code. Adding more code that adds nothing new will not help to make maintaining Virtual TreeView easier.
- In the end this is an open source project and a developer must be found that addresses the issue and pushes code of reasonable quality. Since the ratio of effort and benefit isn't too good, this could get difficult.
